src/components/services/OpenWeather.vue

```diff
@@ -0,0 +1,164 @@
+<template>
+  <div>
+    <div class="card" :class="item.class">
+      <a :href="`https://openweathermap.org/city/` + locationId" :target="item.target" rel="noreferrer">
+        <div class="card-content">
+          <div class="media">
+            <div v-if="image" class="media-left" :class="item.background">
+              <figure class="image is-48x48">
+                <img
+                  :src="`http://openweathermap.org/img/wn/` + image + `@2x.png`"
+                  :alt="conditions"
+                  :title="conditions"
+                />
+              </figure>
+            </div>
+            <div v-if="item.icon" class="media-left">
+              <figure class="image is-48x48">
+                <i style="font-size: 35px" :class="['fa-fw', item.icon]"></i>
+              </figure>
+            </div>
+            <div class="media-content">
+              <p class="title is-4">{{ locationName }}</p>
+              <p class="subtitle is-6">
+                <template v-if="item.subtitle">
+                  {{ item.subtitle }}
+                </template>
+                <template v-else-if="api">
+                  <template v-if="temp !== ''">
+                    {{ temp }}
+                    <span v-if="this.item.units === 'metric'">&deg;C</span>
+                    <span v-else-if="this.item.units === 'imperial'">&deg;F</span>
+                    <span v-else>K</span>
+                  </template>
+                  <template v-else> 
+                    <span class="error">Data could not be retrieved</span>                      
+                  </template>
+                </template>
+              </p>
+            </div>
+          </div>
+          <div class="tag" :class="item.tagstyle" v-if="item.tag">
+            <strong class="tag-text">#{{ item.tag }}</strong>
+          </div>
+        </div>
+      </a>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "OpenWeather",
+  props: {
+    item: Object,
+  },
+  data: () => ({
+    error: false,
+    api: {
+      id: "",
+      name: "",
+      weather: [
+        {
+          main: "",
+          description: "",
+          icon: "",
+        },
+      ],
+      main: {
+        temp: "",
+        humidity: "",
+      },
+    },
+  }),
+  computed: {
+    locationId: function () {
+      if (this.api) {
+        return this.api.id;
+      }
+      return "";
+    },
+    locationName: function () {
+      if (this.api) {
+        return this.api.name;
+      }
+      return "";
```

Comment:
why adding an additional properties when you can access it directly in the template from the component's data (with a default "" value already set)?

Reply:
I would go with `this.api?.name || ''`